perf(RouterLink): use hostListeners for click

PatrickJS opened the pull request:
with a `<router-outlet>` and lots of `router-link` you start to see noticeable lag

vsavkin commented on modules/angular2/src/router/router_link.ts:
```diff
@@ -48,11 +49,12 @@ export class RouterLink {
     this._domEl = elementRef.domElement;
     this._params = StringMapWrapper.create();
-    DOM.on(this._domEl, 'click', (evt) => {
+    this._cancelClick = DOM.onAndCancel(this._domEl, 'click', (evt) => {
```
Why are we using DOM instead of hostListeners?

vsavkin commented on modules/angular2/src/router/router_link.ts:
```diff
@@ -48,11 +49,12 @@ export class RouterLink {
     this._params = StringMapWrapper.create();
-    DOM.on(this._domEl, 'click', (evt) => {
+    this._cancelClick = DOM.onAndCancel(this._domEl, 'click', (evt) => {
       DOM.preventDefault(evt);
```
if you use hostListeners, return false to prevent default.

PatrickJS committed:
perf(RouterLink): use hostListeners for click

with a `<router-outlet>` and lots of `router-link` you start to see
noticeable lag since we’re not removing the listener
